src/generic/Matrix.jl

```diff
@@ -222,9 +205,6 @@ end
 #
 ###############################################################################
 
-function matrix_space(R::AbstractAlgebra.NCRing, r::Int, c::Int; cached::Bool = true)
-   # TODO: the 'cached' argument is ignored and mainly here for backwards compatibility
-   # (and perhaps future compatibility, in case we need it again)
-   T = elem_type(R)
-   return MatSpace{T}(R, r, c)
+function matrix_space(R::AbstractAlgebra.NCRing, r::Int, c::Int)
```

Comment:
I think it is fine to remove the `cached` kwarg for `AA.Generic.matrix_space`.

But I wonder if we should bring it back for `AA. matrix_space` (it was removed in a prior PR, I think) for improved backwards compatibility with existing code calling it? (I guess technically if we don't do that, then this here is a breaking change)

Reply:
I agree, that we should probably add it back and have it for "consistency"?